docs/Use With Phoenix.md

```diff
@@ -0,0 +1,333 @@
+# Using Distillery With Phoenix
```

Comment:
After reviewing this document, let's strip out everything that isn't specifically Phoenix-related, and point people to other documents, or this will end up being a pain to maintain.